src/settings/config.rs

```diff
@@ -0,0 +1,89 @@
+// Config file handling
+
+use std::env;
+
+use serde::Deserialize;
+
+use crate::{dimensions::Dimensions, frame::Frame};
+
+use std::path::{Path, PathBuf};
+
+const CONFIG_FILE: &str = "config.toml";
+
+#[cfg(unix)]
+fn neovide_config_dir() -> PathBuf {
+    let xdg_dirs = xdg::BaseDirectories::with_prefix("neovide").unwrap();
+    xdg_dirs.get_config_home()
+}
+
+#[cfg(windows)]
+fn neovide_config_dir() -> PathBuf {
+    let mut path = dirs::config_dir().unwrap();
+    path.push("neovide");
+    path
+}
+
+pub fn config_path() -> PathBuf {
+    let mut config_path = neovide_config_dir();
+    config_path.push(CONFIG_FILE);
+    config_path
+}
```

Comment:
I think these functions should be in the same place as the ones getting the data path for the save window size, so that it's easy to see where everything is saved. Perhaps you could move those functions from `window_size.rs` to here?

Reply:
I see your point, but looking at both of those things now, I see that they can easily get confused. I guess `PersistentSettings` would need to be renamed. Maybe something like `PersistentState`?

Reply:
Makes sense to me.

And I think it's better to have single serialization format, right now it's json there and toml here. Toml is really preferable for config over json, so it makes sense to store those settings as toml also. But this is suggestion for another refactoring.

And maybe moving those functions here is also better to leave for separate refactoring.

Reply:
Yeah, I agree. No need to do all that in this PR.